code/03_census_and_bed_days.R

```diff
@@ -0,0 +1,868 @@
+            Delay3to14days=sum(Delay3to14days,na.rm=TRUE),Delay2to4weeks=sum(Delay2to4weeks,na.rm=TRUE),
+            Delay4to6weeks=sum(Delay4to6weeks,na.rm=TRUE),Delay6to12weeks=sum(Delay6to12weeks,na.rm=TRUE),
+            Delay3to6months=sum(Delay3to6months,na.rm=TRUE),Delay6to12months=sum(Delay6to12months,na.rm=TRUE),
+            DelayOver12months=sum(DelayOver12months,na.rm=TRUE),DelayOver3days=sum(DelayOver3days,na.rm=TRUE),
+            DelayUnder2wks=sum(DelayUnder2wks,na.rm=TRUE),
+            DelayOver6wks=sum(DelayOver6wks,na.rm=TRUE),DelayOver4wks=sum(DelayOver4wks,na.rm=TRUE),
+            DelayOver2wks=sum(DelayOver2wks,na.rm=TRUE), acute=sum(acute,na.rm=TRUE),
+            gpled=sum(gpled,na.rm=TRUE),notgpled=sum(notgpled,na.rm=TRUE)) %>% 
+  ungroup()
+
+#add files
+Tabs136<-
+  bind_rows(ScotHBlaallreasonsalldelaystotal,ScotHBlaallreasonsincHSCPatFamtotal,ScotHBlaallreasonexcHSCPatFamtotal)
+
+arrange(Tabs136,areaname,age_grp,reas1) # arrange data in order for tables
+
+
+#write_sav(Tabs136,paste0(filepath,"Tabs 1 3 6_R.sav"))
+
+#Tab 4 data - Mean and Media Length of delay IGNORED AS NO LONGER REQUIRED
+## datafile2 is the main census file
+
+#datafile2$level<-"1"
+#datafile2$areaname<-"Scotland"
+#datafile2$age_grp<-"All"
+
+#Create mean/median delay for following high level groupings in reason_grp_high_level
+#Code 9 / Health and Social Care Reasons/Patient/Carer/Family-related reasons
+
+#Scotlandhighlevelgrouping <- datafile2 %>% 
+#  group_by(year,monthflag,level,areaname,age_grp,reason_grp_high_level) %>% 
+#  summarise(mean_delay=mean(delay_at_census),
+#            median_delay=median(delay_at_census)) %>% 
+#  ungroup()
+
+
+#write_sav(Scotlandhighlevelgrouping,paste0(filepath,"Scotland high level groupings.sav")) # save out file
+
+#code
+#datafile3<-datafile2 %>% mutate(reason_grp_high_level<-"All")
+
+
+#Scotlandallreasons <- datafile3 %>% 
+#  group_by(year,monthflag,level,areaname,age_grp,reason_grp_high_level) %>% 
+#  summarise(mean_delay=mean(delay_at_census),
+#            median_delay=median(delay_at_census)) %>% 
+#  ungroup()
+
+
+#Recode Reas2 to group Code 9 non-AWI patients into 'Other Code9'.
+
+
+#datafile3<-datafile3 %>% mutate(reason_grp=
+#              if_else(delay_description!="Adults with Incapacity Act" & reason_grp_high_level=="Code 9", "Other code 9 reasons(not AWI)",reason_grp))
+
+
+#Create Mean / Median delay for sub groupings in reas1 ( which is reason_grp_high_level)
+
+#reas1=reas2
+#datafile4<-datafile3 %>% mutate(reason_grp_high_level=reason_grp)
+
+#Scotlandsublevelgroupings <- datafile4 %>% 
+#  group_by(year,monthflag,level,areaname,age_grp,reason_grp_high_level) %>% 
+#  summarise(mean_delay=mean(delay_at_census),
+#            median_delay=median(delay_at_census)) %>% 
+#  ungroup()
+
+###12. Now get detailed reason breakdown ----
+#get main census file ( datafile2)
+datafile5<-datafile2
+
+datafile5$level<-"1"
+datafile5$areaname<-"Scotland"
+datafile5$age_grp<-"All"
```

Comment:
See "Incremental Data-Frame Names" section in general feedback.
```suggestion
datafile %>% mutate(level = "1", areaname = "Scotland", age_grp = "All")
```